File: member-service/src/test/java/com/hedvig/memberservice/services/CashbackServiceTest.java
package com.hedvig.memberservice.services;

import com.hedvig.memberservice.web.dto.CashbackOption;
import org.junit.Test;

import java.util.Optional;
import java.util.UUID;

import static org.junit.Assert.assertFalse;

public class CashbackServiceTest {


    @Test
    public void getCashbackOption_RETURNS_Empty_WHEN_CashbackIdNotFound() throws Exception {
        CashbackService service = new CashbackService();

        final Optional<CashbackOption> acctual = service.getCashbackOption(UUID.randomUUID());

        assertFalse(acctual.isPresent());
    }

}

File: member-service/src/test/java/com/hedvig/memberservice/services/CashbackServiceTest.kt
package com.hedvig.memberservice.services

import com.hedvig.localization.service.LocalizationService
import com.hedvig.memberservice.aggregates.PickedLocale
import org.junit.Assert
import org.junit.Test
import java.util.*

class CashbackServiceTest {
    @Test
    fun cashbackOption_RETURNS_Empty_WHEN_CashbackIdNotFound() {
            val service = CashbackService(MockLocalizationService())
            val acctual = service.getCashbackOption(UUID.randomUUID(), PickedLocale.sv_SE)
            Assert.assertFalse(acctual.isPresent)
        }
}

class MockLocalizationService: LocalizationService {
    override fun getText(locale: Locale?, key: String) = "mock"
}
